datadog/resource_datadog_logs_index.go

```diff
@@ -0,0 +1,168 @@
+package datadog
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/zorkian/go-datadog-api"
+	"strings"
+)
+
+var indexSchema = map[string]*schema.Schema{
+	"name": {Type: schema.TypeString, Required: true},
+	"filter": {
+		Type:     schema.TypeList,
+		Required: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"query": {Type: schema.TypeString, Required: true},
+			},
+		},
+	},
+	"exclusion_filter": {
+		Type:     schema.TypeList,
+		Required: true,
+		Elem: &schema.Resource{
+			Schema: exclusionFilterSchema,
+		},
+	},
+}
+
+var exclusionFilterSchema = map[string]*schema.Schema{
+	"name":       {Type: schema.TypeString, Optional: true},
+	"is_enabled": {Type: schema.TypeBool, Optional: true},
+	"filter": {
+		Type:     schema.TypeList,
+		Required: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"query":       {Type: schema.TypeString, Optional: true},
+				"sample_rate": {Type: schema.TypeFloat, Optional: true},
+			},
+		},
+	},
+}
+
+func resourceDatadogLogsIndex() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogLogsIndexCreate,
+		Update: resourceDatadogLogsIndexUpdate,
+		Read:   resourceDatadogLogsIndexRead,
+		Delete: resourceDatadogLogsIndexDelete,
+		Exists: resourceDatadogLogsIndexExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: indexSchema,
+	}
+}
+
+func resourceDatadogLogsIndexCreate(d *schema.ResourceData, meta interface{}) error {
+	return resourceDatadogLogsIndexUpdate(d, meta)
+}
+
+func resourceDatadogLogsIndexRead(d *schema.ResourceData, meta interface{}) error {
+	ddIndex, err := meta.(*datadog.Client).GetLogsIndex(d.Id())
+	if err != nil {
+		return err
+	}
+	if err = d.Set("name", ddIndex.GetName()); err != nil {
+		return err
+	}
+	if err = d.Set("filter", buildTerraformFilter(ddIndex.Filter)); err != nil {
+		return err
+	}
+	if err = d.Set("exclusion_filter", buildTerraformExclusionFilters(ddIndex.ExclusionFilters)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceDatadogLogsIndexUpdate(d *schema.ResourceData, meta interface{}) error {
+	ddIndex, err := buildDatadogIndex(d)
+	if err != nil {
```

Comment:
It would be great if this could explicitly warn the user that creating new indexes is not allowed when 404 is returned here.

Reply:
will do.